pkg/services/ngalert/models/instance.go

```diff
@@ -48,6 +62,24 @@ func (i InstanceStateType) IsValid() bool {
 		i == InstanceStateError
 }
 
+// IsValid checks that the value of InstanceCauseType is a valid string.
+func (i InstanceCauseType) IsValid() bool {
+	return i == InstanceNoCause ||
+		i == InstanceCauseFiring ||
+		i == InstanceCauseError ||
+		i == InstanceCauseNoData
+}
+
+// validateCurrentStateAndCurrentPendingState checks that the possible combinations of CurrentState and
+// CurrentCause are valid.
+func validateCurrentStateAndCurrentPendingState(cState InstanceStateType, cCause InstanceCauseType) bool {
```

Comment:
I don't think this is required. We set the combination of cause and state, not the user. I think this validation is better left for tests, and then we can remove this.